src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java

```diff
@@ -132,21 +131,24 @@ public static List<NoticeListItem> convertToNoticeList(TransactionListResponse t
                 .toList();
     }
 
-    public static TransactionListItem convertTransactionListItem(BizEventsViewUser viewUser, List<BizEventsViewCart> listOfCartViews) {
+    public static TransactionListItem convertTransactionListItem(BizEventsViewUser viewUser, BizEventsViewCart bizEventsViewCart, BizEventsViewGeneral bizEventsViewGeneral) {
         AtomicReference<BigDecimal> totalAmount = new AtomicReference<>(BigDecimal.ZERO);
+        BigDecimal amountExtracted = new BigDecimal(bizEventsViewCart.getAmount());
+        totalAmount.updateAndGet(v -> v.add(amountExtracted));
+        
+        /*
         for (BizEventsViewCart bizEventsViewCart : listOfCartViews) {
             BigDecimal amountExtracted = new BigDecimal(bizEventsViewCart.getAmount());
             totalAmount.updateAndGet(v -> v.add(amountExtracted));
-        }
+        }*/
 
         return TransactionListItem.builder()
-                .transactionId(viewUser.getTransactionId())
-                .payeeName(listOfCartViews.size() > 1 ? payeeCartName : listOfCartViews.get(0).getPayee().getName())
-                .payeeTaxCode(listOfCartViews.size() > 1 ? "" : listOfCartViews.get(0).getPayee().getTaxCode())
-                // PAGOPA-1763: the amount value must be returned only if it is not a cart type transaction
-                .amount(listOfCartViews.size() > 1 && BooleanUtils.isTrue(viewUser.getIsDebtor()) ? null : totalAmount.get().setScale(2, RoundingMode.UNNECESSARY).toString())
+                .transactionId(viewUser.getId().replace("-p", "").replace("-d", "")) // eventId stripped of the -d or -p suffix
```

Comment:
directly remove the last 2 characters.
The replace, in addition to being done twice, may not be safe if that character exists within the Id.
and just leave the comment that explains and gives context